src/main/kotlin/com/sourcegraph/cody/chat/ui/ContextFilesPanel.kt

```diff
@@ -54,53 +61,91 @@ class ContextFilesPanel(
       val accordionSection = AccordionSection("Read ${filesAvailableInEditor.size} files")
       accordionSection.isOpaque = false
       accordionSection.border = EmptyBorder(margin)
-      filesAvailableInEditor.forEachIndexed { index, file: VirtualFile ->
-        val filePanel = createFileWithLinkPanel(file)
+      filesAvailableInEditor.forEachIndexed { index, file: Pair<VirtualFile, ContextFileFile?> ->
+        val filePanel = createFileWithLinkPanel(file.first, file.second)
         accordionSection.contentPanel.add(filePanel, index)
       }
       add(accordionSection, BorderLayout.CENTER)
     }
   }
 
   @RequiresEdt
-  private fun createFileWithLinkPanel(file: VirtualFile): JPanel {
+  private fun createFileWithLinkPanel(
+      file: VirtualFile,
+      contextFileFile: ContextFileFile?
+  ): JPanel {
     val projectRelativeFilePath = file.path.removePrefix(project.basePath ?: "")
     val anAction =
         object : DumbAwareAction() {
           override fun actionPerformed(anActionEvent: AnActionEvent) {
-            logger.info(
-                "Opening a file from the used context (projectRelativeFilePath=$projectRelativeFilePath, file=$file)")
-            FileEditorManager.getInstance(project).openFile(file, /*focusEditor=*/ true)
+            val logicalLine = contextFileFile?.range?.start?.line ?: 0
+            OpenFileDescriptor(project, file, logicalLine, /* logicalColumn= */ 0)
+                .navigate(/* requestFocus= */ true)
           }
         }
-    val goToFile = AnActionLink("@$projectRelativeFilePath", anAction)
+    val actionText = getActionTextForFileLinkAction(contextFileFile, projectRelativeFilePath)
+    val goToFile = AnActionLink(actionText, anAction)
+    goToFile.toolTipText = actionText
     val panel = JPanel(BorderLayout())
     panel.isOpaque = false
     panel.border = JBUI.Borders.emptyLeft(3)
     panel.add(goToFile, BorderLayout.PAGE_START)
     return panel
   }
 
+  private fun getActionTextForFileLinkAction(
+      contextFileFile: ContextFileFile?,
+      projectRelativeFilePath: String
+  ): String {
+    val (startLine, endLine) = contextFileFile?.range?.intellijRange() ?: Pair(null, null)
+
+    return buildString {
+      append("@$projectRelativeFilePath")
+      if (startLine != null && endLine != null) {
```

Comment:
You resolved my comment but problem I mentioned is not addressed I think.

Reply:
fixed now